# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
답변:
1. A/B 테스트란?

A/B 테스트는 같은 목표(전환, 클릭, 구매, 체류 시간 등)를 두고, 서로 다른 버전 (A와 B)를 사용자에게 무작위로 노출한 뒤 성과 차이가 우연인지(노이즈) 실제 효과인지를 통계적으로 판단하는 실험을 말한다.

다시말해, 실제 사용자 행동 데이터를 이용한 의사 결정의 표준 도구이다. 따라서 핵심은 '무작위 배정', '동일 조건', '사전 정의된 지표'가 된다.

A(Control): 현재 버전, 기준

B(Alternative): 바꾸고 싶은 새로운 버전(문구, 버튼 색, 가격 표시, 레이아웃, 추천 순서 등)

2. A/B 테스트의 예시

상황: 암/노화 연구실에서 신규 논문을 발표했고, 공동 연구 제안, 학생/포닥 지원, 데이터 요청을 늘리고 싶다.

가설: "논문의 biological significance를 먼저 강조하면, 연구자들의 문의 전환률이 증가할 것이다.

2-1. 실험 설계

A: 홈페이지 상단의 기술 중심 설명(We identified a novel mTORC1-FAM120A axis regulating SASP)

B: 홈페이지 상단의 임상적 의미 제시(Targeting mTORC1-FAM120A rewires SASP and alters tumor-stroma communication)

2-2. 핵심 지표 (KPI)

논문 페이지 상의 Contact/Collaboration 문의 전환률

+ 보조 지표: 페이지 체류 시간, PDF 다운로드 후 이탈률

2-3. 유의 사항

2-3-1. 초두 효과

> 초두 효과(Primacy effect): 처음 본 경험이 이후 판단을 강하게 좌우하는 현상이다. 기존 방식에 익숙한 유저는 새 방식을 불편하다고 느껴 성과가 떨어질 수 있고, 반대로 신규 유저는 새 방식이 더 직관적일 수 있다. 즉, 전체 평균만 보면 효과가 왜곡될 수 있음을 고려해야 한다.

> 신규 유저 vs 기존 유저 분리 분석(cohort split), 런인(run-in) 기간, 적응 기간 고려, 교차 오염 방지(sticky assignment) 등을 활용해서 해결할 수 있다.

연구자는 첫 문단에서 논문의 급이나 핵심 내용을 판단하는 경우가 많고, 연구실의 기존 연구 방향에 익숙한 사람은 mechanistic detail, molecular mechanism을 선호하지만, 신규 분야 연구자나 MD phd는 biological/clinical relevance를 선호하는 경향이 있다. 따라서 분리 분석이 필요하다.

2-3-2. 표본 수 & 실험 기간

연구실 홈페이지의 특성상 하루 방문자 수가 적고, noise가 커진다. 반면, 학회 시즌이나 논문 발표 직후에는 트래픽이 급증하기 때문에 confounder가 발생한다.

2-3-3. 변수 통제

문구 + figure와 title 배치 등을 동시에 바꾸면 어떤 요소가 어떤 효과를 냈는지 알 수 없다.









---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [Effi-ending] 공모전 카드 정보 밀도 조정에 따른 탐색 효율 개선 실험 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 공모전 참여 의향은 있으나, 현제 메인 페이지에서 카드 정보 과부화로 핵심 공모전을 빠르게 선별하지 못하는 유저 |
| **Observation** | **데이터 근거** <br> • 메인 페이지 체류 시간은 평균 12초로 짧음 <br> • 공모전 카드 클릭 유저의 상세 진입률은 45%로 준수하나, 전체 카드 노출 대비 클릭률은 12%로 낮음. <br> • 유저 인터뷰 결과, "정보가 많아 처음에는 다 비슷하고 고르기가 어렵다."는 피드백 |
| **Problem Statement** | 메인 화면에 채도와 명도가 높은 디자인의 서로 다른 공모전 정보(마감일, 세부 주제, 제목)가 동시에 노출되어 인지 부하가 높음 |
| **Hypothesis** | ** 공모전 카드 메인 화면에서 결정적인 공모전 2-3개를 우선 노출하고, 나머지는 이어지는 페이지/스크롤 이후에 제공할 경우 카드 클릭률과 상세 진입률이 개선될 것이다.**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/BXj532dv/Screenshot-2026-02-03-at-2-23-04-PM.png" width="100%"><br>
      <sub>기존안: 동시 과다 노출</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/568080zm/Screenshot-2026-02-03-at-2-28-48-PM.png" width="100%"><br>
      <sub>변경안: Top Picks 3개 고정</sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   
또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:

p1, p2는 두 그룹의 전환율로 잡을 수 있다(e.g. 기존 CVR이 25% 라면  p1 = 0.25). 이 둘의 차이 p1-p2가 검출하고 싶은 효과 크기(MDE)이다.

p(1-p)는 비율의 분산이 된다. 베르누이 시행에서의 분산이며, 전환 여부가 성공/실패의 이진 변수이기 때문이다.

분모 (p1-p2)^2 은 효과 크기의 제곱으로, 차이가 작을 수록 필요한 표본 수가 급증하게 된다는 직관과 일치한다.

결과로 나온 n은 각 그룹 당 필요한 표본 개수를 의미한다. (총 필요 표본 수는 2n)

유의 수준이 α = 0.05, 검정력을 80%로 가정했으므로 코드는 다음과 같다.

In [19]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.

import math
from scipy.stats import norm

def sample_size(p1, p2, alpha=0.05, power=0.8):

  # z 값 계산
  z_alpha_over_2 = norm.ppf(1 - alpha/2)
  z_beta = norm.ppf(power)

  # 분산 항
  var = p1 * (1 - p1) + p2 * (1 - p2)

  # 효과 크기
  effect_size = (p1 - p2) ** 2

  # 표본 수 계산
  n = ((z_alpha_over_2 + z_beta) ** 2 * var) / effect_size


  # 결과 출력(표본 개수는 올림 처리)
  return math.ceil(n)

In [23]:
# 예시 데이터
p1 = 0.25 # 기존안의 CVR
p2 = 0.30 # 변경안의 CVR
n = sample_size(p1, p2)
print("총 필요 표본 수(minimum sample size, n):", n)

총 필요 표본 수(minimum sample size, n): 1248


## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수?  

600명/일

산출된 최소 test users 수?

n = 1248
  
그래서 필요한 최소 기간은?

2496/600 = 4.16 -> 5일

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 카드 클릭률(Card CTR)
> * **Secondary Metric (보조지표):** 상세 페이지 체류 시간, 회원 가입
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률(Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **총 1248 명** | 그룹당 인원수 명시 |
| **Required Period** | **최소 5일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off :**    
    상위 2-3개 공모전만 우선 노출되면서, 첫 화면에서 다양한 선택지가 줄어 일부 유저는 정보가 부족하다고 느낄 가능성이 있다.

* **Andon (중단 기준) :**
    1. 주지표가 대조군 대비 5% 이상 하락하고, 통계적으로 유의한 방향으로 경향이 지속될 경우 중단, Bounce Rate가 대조군 대비 10%p 이상 증가가 2일 연속 관찰될 경우 중단

    2. UI 변경으로 인한 페이지 오류(로딩 실패, 클릭 불가 등)또는 CS/불만 보고 급증할 경우 즉시 중단
  

---

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 검정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 카이제곱검정이고,  

이유는 본 실험을 메인 홈 검색 모듈의 위치와 클릭 여부라는 두 범주형 변수 간의 독립성 여부를 검정하는 문제로 볼 때 카이 제곱을 이용한 독립성 검정을 적용할 수 있기 때문이다.

<br>

In [ ]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

카이제곱 통계량: 26.9884
P-value: 0.0000002047
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  

A(Control): 기존안, 메인 화면에 다수의 공모전 카드 + 정보 노출 과다

B(Alternative): 메인 화면에 결정적 공모전 2-3개만 우선 노출, 나머지는 스크롤 이후 제공

귀무가설 ($H_0$): 공모전 카드 노출 방식(기존안 vs 변경안)은 카드 클릭 여부(CTR)에 영향을 주지 않는다. 즉, 두 그룹의 카드 클릭률은 동일하다.

$$H_0: p_A = p_B$$

대립가설 ($H_1$): 메인 화면에서 결정적 공모전 2-3개를 우선 노출할 경우, 기존안 대비 변경안의 클릭률이 증가한다. $$H_1: p_A < p_B$$

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 one-sided z-test 이고,  

이유는 카드 클릭 여부는 클릭(1), 미클릭(0)의 이진 변수이며, CTR은 이진 변수의 비율이 된다. 표본 수가 충분히 크면 표본 비율은 정규 분포로 근사 가능하며(CLT) chi-square test와 달리 모비율의 비율 차이를 검정하는 것이 가능하기 때문이다.

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [24]:
### A3-4. 코드작성

import math
from scipy.stats import norm

# 가정 데이터 (CTR)

# 메인 페이지 방문 수
nA = 5000   # 기존안 노출 수
nB = 5000   # 변경안 노출 수

# 카드 클릭 수
clickA = 600   # CTR_A = 0.12
clickB = 750   # CTR_B = 0.15

alpha = 0.05  # 유의수준

# 두 모비율 Z-test (단측)
# H0: pA = pB
# H1: pB > pA

pA_hat = clickA / nA
pB_hat = clickB / nB

# pooled proportion (H0 가정)
p_pool = (clickA + clickB) / (nA + nB)

# 표준오차
se = math.sqrt(p_pool * (1 - p_pool) * (1/nA + 1/nB))

# Z 통계량
z = (pB_hat - pA_hat) / se

# 단측 p-value
p_value = 1 - norm.cdf(z)

print(f"A CTR = {pA_hat:.4f} ({pA_hat*100:.2f}%)")
print(f"B CTR = {pB_hat:.4f} ({pB_hat*100:.2f}%)")
print(f"z = {z:.4f}")
print(f"p-value = {p_value:.6f}")

if p_value < alpha:
    print("결론: 귀무가설 기각 → 변경안의 CTR이 유의하게 높다")
else:
    print("결론: 귀무가설 기각 실패 → CTR 증가가 유의하지 않다")


A CTR = 0.1200 (12.00%)
B CTR = 0.1500 (15.00%)
z = 4.3895
p-value = 0.000006
결론: 귀무가설 기각 → 변경안의 CTR이 유의하게 높다


**결과해석**

답변: 두 모비율에 대한 one-sided z-test 결과, p-value가 유의수준 0.05보다 작아 귀무 가설을 기각하였다. 이는 메인 화면에서 결정적인 공모전 2-3개를 우선 노출하는 변경안이 기존안 대비 카드 클릭률을 유의하게 증가시킨다는 근거가 된다.

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**